### PR TITLE
prisoner's dilemma

### LINKED ISSUE #4: update earth contract to use prisoners's dilemma
## problem

linear game is boring.

## solution
introduce prisoners dilemma for trades with this matrix:

IMG_20190717_104437

## payment conditions
- contracts and tests
- reviewer by owners

## roles
reviewer: @vrde 20%
gardner / worker @johannbarbie 80

### PR DESCRIPTION
**Pull request by johba37**
resolves https://github.com/social-dist0rtion-protocol/planet-a-contracts/issues/4

## How does Citizen A signal collaborate/defect?

**collaborate:** increase CO2Locked in passport receipt by 1.
**defect:** sign passport receipt without any changes.

## How does Citizen B signal collaborate/defect?

**collaborate:** citizen B attaches a signed input with goellars (allowance).
**defect:** citizen B provides no allowance, only sign passport input.

### CODE REVIEW

**Comment by vrde** [comment]
Looks good, but I'd invert the logic so if Citizen B doesn't have money (Göllars to sign an input) then they **collaborate** by default. I'd also change the rules for Citizen A to collaborate/defect. What do you think?

## How does Citizen A signal collaborate/defect?

**collaborate:** sign passport receipt without any changes.
**defect:** increase CO2Locked in passport receipt by 1.

## How does Citizen B signal collaborate/defect?

**collaborate:** citizen B provides no allowance, only sign passport input.
**defect:** citizen B attaches a signed input with goellars (allowance).

File: test/earth.js [hunk 1]
```diff
@@ -48,7 +48,164 @@ contract('Earth Contract', (accounts) => {
...
+    // fund earth
+    await goellars.transfer(earth.address, '500000000000000000000');
+    await co2.transfer(earth.address, totalCo2);
+
+    // print passports for citizens
+    await countryA.mint(citizenA, passportA);
+    await countryB.mint(citizenB, passportB);
+
+    // citizen A sharing signed receipt through QR code
+    const dataAfterColaborate = '0000000000000000000000000000000000000000000000000000000100001388';
```
Comment (vrde) [inline test/earth.js]: `dataAfterCollaborate`, double `l`

**Comment by johba37** [comment]
@vrde done :)

**Review by vrde — APPROVED** [review]
Tried on the Planet A burner wallet and it works!

**Pull request merged**